src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/References/ReferenceCleanupService.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Exceptions;
+using Microsoft.VisualStudio.LanguageServices.ExternalAccess.ProjectSystem.Api;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.References
+{
+    [Export(typeof(IProjectSystemReferenceCleanupService))]
+    internal class ReferenceCleanupService : IProjectSystemReferenceCleanupService
+    {
+        private static readonly Dictionary<ProjectSystemReferenceType, AbstractReferenceHandler> s_mapReferenceTypeToHandler =
+            new Dictionary<ProjectSystemReferenceType, AbstractReferenceHandler>()
+            {
+                { ProjectSystemReferenceType.Project, new ProjectReferenceHandler() },
+                { ProjectSystemReferenceType.Package, new PackageReferenceHandler() },
+                { ProjectSystemReferenceType.Assembly, new AssemblyReferenceHandler() }
+            };
+
+        private readonly Lazy<IProjectService2> _projectService;
+        protected IProjectService2 ProjectService => _projectService.Value;
+
+        [ImportingConstructor]
+        public ReferenceCleanupService(IProjectServiceAccessor projectServiceAccessor)
+        {
+            _projectService = new Lazy<IProjectService2>(
+                () => (IProjectService2)projectServiceAccessor.GetProjectService(),
+                LazyThreadSafetyMode.PublicationOnly);
+        }
+
+        public async Task<ImmutableArray<ProjectSystemReferenceInfo>> GetProjectReferencesAsync(string projectPath, CancellationToken cancellationToken)
+        {
+            List<ProjectSystemReferenceInfo> references;
+
+            try
+            {
+                var activeConfiguredProject = await GetActiveConfiguredProjectByPathAsync(projectPath, cancellationToken);
+
+                references = await GetAllReferencesInConfiguredProjectAsync(activeConfiguredProject, cancellationToken);
+            }
+            catch
+            {
+                throw new InvalidProjectFileException();
+            }
+
+            return references.ToImmutableArray();
+        }
+
+        private async Task<ConfiguredProject> GetActiveConfiguredProjectByPathAsync(string projectPath, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var unconfiguredProjectPath = ProjectService.GetLoadedProject(projectPath);
+
+            if (unconfiguredProjectPath is null)
+            {
+                throw new InvalidProjectFileException();
+            }
+
+            var activeConfiguredProject = await unconfiguredProjectPath.GetSuggestedConfiguredProjectAsync();
```

Comment:
With this change, the exception changes from `InvalidProjectFileException` to `NullReferenceException`. You cannot `await` a `null` value.

Reply:
Ah. I see. Then, it would need to be more annoying, like:
```
await ProjectService.GetLoadedProject(projectPath)?.GetSuggestedConfiguredProjectAsync()?.AsTask() ?? Task.CompletedTask;
```

Fun times with C#. 😝